tmp/test_instagram.rb

```diff
@@ -0,0 +1,166 @@
+  user = client.user
+  html = "<h1>#{user.username}'s likes feed</h1>"
+end
+
+get "/user_media_feed" do
+  client = Instagram.client(:access_token => session[:access_token])
+  user = client.user
+  html = "<h1>#{user.username}'s media feed</h1>"
+
+  page_1 = client.user_media_feed(777)
+  page_2_max_id = page_1.pagination.next_max_id
+  page_2 = client.user_recent_media(777, :max_id => page_2_max_id ) unless page_2_max_id.nil?
+  html << "<h2>Page 1</h2><br/>"
+  for media_item in page_1
+    html << "<img src='#{media_item.images.thumbnail.url}'>"
+  end
+  html << "<h2>Page 2</h2><br/>"
+  for media_item in page_2
+    html << "<img src='#{media_item.images.thumbnail.url}'>"
+  end
+  html
+end
+
+get "/location_recent_media" do
+  client = Instagram.client(:access_token => session[:access_token])
+  html = "<h1>Media from the Instagram Office</h1>"
+  for media_item in client.location_recent_media(514276)
+    html << "<img src='#{media_item.images.thumbnail.url}'>"
+  end
+  html
+end
+
+get "/media_search" do
+  client = Instagram.client(:access_token => session[:access_token])
+  html = "<h1>Get a list of media close to a given latitude and longitude</h1>"
+  for media_item in client.media_search("37.7808851","-122.3948632")
+    html << "<img src='#{media_item.images.thumbnail.url}'>"
+  end
+  html
+end
+
+get "/media_popular" do
+  client = Instagram.client(:access_token => session[:access_token])
+  html = "<h1>Get a list of the overall most popular media items</h1>"
+  for media_item in client.media_popular
+    html << "<img src='#{media_item.images.thumbnail.url}'>"
+  end
+  html
+end
+
+get "/user_search" do
+  client = Instagram.client(:access_token => session[:access_token])
+  html = "<h1>Search for users on instagram, by name or usernames</h1>"
+  for user in client.user_search("instagram")
+    html << "<li> <img src='#{user.profile_picture}'> #{user.username} #{user.full_name}</li>"
+  end
+  html
+end
+
+get "/location_search" do
+  client = Instagram.client(:access_token => session[:access_token])
+  html = "<h1>Search for a location by lat/lng with a radius of 5000m</h1>"
+  for location in client.location_search("48.858844","2.294351","5000")
+    html << "<li> #{location.name} <a href='https://www.google.com/maps/preview/@#{location.latitude},#{location.longitude},19z'>Map</a></li>"
+  end
+  html
+end
+
+get "/location_search_4square" do
+  client = Instagram.client(:access_token => session[:access_token])
+  html = "<h1>Search for a location by Fousquare ID (v2)</h1>"
+  for location in client.location_search("3fd66200f964a520c5f11ee3")
+    html << "<li> #{location.name} <a href='https://www.google.com/maps/preview/@#{location.latitude},#{location.longitude},19z'>Map</a></li>"
+  end
+  html
+end
+
+get "/tags" do
+  client = Instagram.client(:access_token => session[:access_token])
+  html = "<h1>Search for tags, get tag info and get media by tag</h1>"
+  tags = client.tag_search('cat')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.